src/screens/CreatePost/elements/useHastagMention.js

```diff
@@ -6,14 +6,14 @@
   const [formattedText, setFormattedText] = React.useState(initial);
   const [hashtags, setHashtags] = React.useState([]);
 
-  const handleStateHashtag = (text, setHashtagState = null) => {
-    const hashtagMention = handleHastagMention(text, hashtags, setHashtagState);
+  const handleStateHashtag = (text) => {
+    const hashtagMention = handleHastagMention(text, hashtags);
     setFormattedText(hashtagMention);
   };
 
-  const updateHashtag = (text, hashtagsProp, setHashtagState = null) => {
+  const updateHashtag = (text, hashtagsProp) => {
     setHashtags(hashtagsProp);
-    setFormattedText(handleHastagMention(text, hashtagsProp, setHashtagState));
+    setFormattedText(handleHastagMention(text, hashtagsProp));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similarly, the `updateHashtag` function has also been simplified by removing the optional `setHashtagState` parameter. This change is consistent with the previous one and helps maintain simplicity and clarity in the code.

```diff
-  const updateHashtag = (text, hashtagsProp, setHashtagState = null) => {
-    setFormattedText(handleHastagMention(text, hashtagsProp, setHashtagState));
+  const updateHashtag = (text, hashtagsProp) => {
+    setFormattedText(handleHastagMention(text, hashtagsProp));
   };
```
Overall, these changes seem to improve the readability and maintainability of the code by reducing unnecessary complexity.


<!-- This is an auto-generated comment by OSS CodeRabbit -->